### PR TITLE
Fixing kafkaSink buffer config props

### PR DESCRIPTION
**Pull request by jatinder85**
Updated kafkaSink with correct buffer properties for the producer.

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

📝 Please visit https://github.com/snowplow/snowplow/wiki/CLA to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

**Comment by jatinder85** [comment]
I have signed the agreement.

**Comment by snowplowcla** [comment]
Confirmed! @jatinder85 has signed the Individual Contributor License Agreement. Thanks so much

File: 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/sinks/KafkaSink.scala [hunk 1]
```diff
@@ -57,11 +57,12 @@ class KafkaSink(config: CollectorConfig, inputType: InputType.InputType) extends
     props.put("bootstrap.servers", config.kafkaBrokers)
     props.put("acks", "all")
     props.put("retries", "0")
-    props.put("batch.size", config.byteLimit.toString)
+    props.put("buffer.memory", config.byteLimit.toString)
+    props.put("batch.size", config.recordLimit.toString)
```
Comment (simplesteph) [inline 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/sinks/KafkaSink.scala]: very important fix here, can we please merge ASAP?

Comment (BenFradet) [inline 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/sinks/KafkaSink.scala]: mmmh, the `record-limit` configuration represents a number of records, `batch.size` is a number of bytes, I don't think it really applies here.

**Comment by BenFradet** [comment]
Integrated into #3274, thanks!

**Pull request closed without merging**